src/libs/actions/TaxRate.ts

```diff
@@ -119,10 +183,289 @@ function clearTaxRateError(policyID: string, taxID: string, pendingAction?: Pend
+    const originalTaxes = {...policy?.taxRates?.taxes};
+
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: taxesIDsToUpdate.reduce<TaxRateEnabledMap>((acc, taxID) => {
+                            acc[taxID] = {
+                                isDisabled: !isEnabled,
+                                pendingFields: {isDisabled: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE},
+                                errorFields: {isDisabled: null},
+                            };
+                            return acc;
+                        }, {}),
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: taxesIDsToUpdate.reduce<TaxRateEnabledMap>((acc, taxID) => {
+                            acc[taxID] = {pendingFields: {isDisabled: null}, errorFields: {isDisabled: null}};
+                            return acc;
+                        }, {}),
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: taxesIDsToUpdate.reduce<TaxRateEnabledMap>((acc, taxID) => {
+                            acc[taxID] = {
+                                isDisabled: !!originalTaxes[taxID].isDisabled,
+                                pendingFields: {isDisabled: null},
+                                errorFields: {isDisabled: ErrorUtils.getMicroSecondOnyxError('workspace.taxes.errors.updateFailureMessage')},
+                            };
+                            return acc;
+                        }, {}),
+                    },
+                },
+            },
+        ],
+    };
+
+    const parameters = {
+        policyID,
+        taxFieldsArray: JSON.stringify(taxesIDsToUpdate.map((taxID) => ({taxCode: taxID, enabled: isEnabled}))),
+    } satisfies SetPolicyTaxesEnabledParams;
+
+    API.write(WRITE_COMMANDS.SET_POLICY_TAXES_ENABLED, parameters, onyxData);
+}
+
+type TaxRateDeleteMap = Record<
+    string,
+    | (Pick<TaxRate, 'pendingAction'> & {
+          errors: OnyxCommon.Errors | null;
+      })
+    | null
+>;
+
+function deletePolicyTaxes(policyID: string, taxesToDelete: string[]) {
+    const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
+    const policyTaxRates = policy?.taxRates?.taxes;
+
+    if (!policyTaxRates) {
```

Comment:
Same above

Reply:
I can add it, but it would result in optional chaining below:
```
taxNames: JSON.stringify(taxesToDelete.map((taxID) => policyTaxRates?.[taxID]?.name)),
```